icloud-auth/src/client.rs

```diff
@@ -526,11 +669,13 @@ impl AppleAccount {
                 );
             });
 
-        headers.insert(
-            "Content-Type",
-            HeaderValue::from_str("text/x-xml-plist").unwrap(),
-        );
-        headers.insert("Accept", HeaderValue::from_str("text/x-xml-plist").unwrap());
+        if !sms {
+            headers.insert(
+                "Content-Type",
+                HeaderValue::from_str("text/x-xml-plist").unwrap(),
+            );
+            headers.insert("Accept", HeaderValue::from_str("text/x-xml-plist").unwrap());
+        }
         headers.insert("User-Agent", HeaderValue::from_str("Xcode").unwrap());
```

Comment:
This User-Agent should maybe get customizable? omnisette also hardcodes one iirc